src/Agent.Sdk/MountVolume.cs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace Agent.Sdk
+{
+    public class MountVolume
+    {
+        public MountVolume()
+        {
+
+        }
+
+        public MountVolume(string sourceVolumePath, string targetVolumePath, bool readOnly = false)
+        {
+            this.SourceVolumePath = sourceVolumePath;
+            this.TargetVolumePath = targetVolumePath;
+            this.ReadOnly = readOnly;
+        }
+
+        public MountVolume(string fromString)
+        {
+            ParseVolumeString(fromString);
+        }
+
+        private static Regex autoEscapeWindowsDriveRegex = new Regex(@"(^|:)([a-zA-Z]):(\\|/)", RegexOptions.Compiled);
+        private string AutoEscapeWindowsDriveInPath(string path)
+        {
+
+            return autoEscapeWindowsDriveRegex.Replace(path, @"$1$2\:$3");
+        }
+
+        private void ParseVolumeString(string volume)
+        {
+            ReadOnly = false;
+            SourceVolumePath = null;
+
+            string readonlyToken = ":ro";
+            if (volume.ToLower().EndsWith(readonlyToken))
+            {
+                ReadOnly = true;
+                volume = volume.Remove(volume.Length-readonlyToken.Length);
+            }
+            // for completeness, in case someone explicitly added :rw in the volume mapping, we should strip it as well
+            string readWriteToken = ":rw";
+            if (volume.ToLower().EndsWith(readWriteToken))
+            {
+                ReadOnly = false;
+                volume = volume.Remove(volume.Length-readWriteToken.Length);
+            }
+
+            if (volume.StartsWith(":"))
+            {
+                volume = volume.Substring(1);
+            }
+
+            var volumes = new List<string>();
+            // split by colon, but honor escaping of colons
+            var volumeSplit = AutoEscapeWindowsDriveInPath(volume).Split(':');
```

Comment:
Will this be OK for cross OS?

Reply:
Yes, colons are not allowed as path chars in unix